File: fixtures/scale/swift/large/Families/Atbash/AtbashDescriptor.swift
struct AtbashDescriptor: FamilyDescriptor {
    var family: String { "atbash" }
    var suite: String { "classical" }
    var kind: CipherKind { .letter }

    func cipher() -> Cipher {
        return AtbashCipher()
    }

    func vectors() -> [TestVector] {
        return AtbashVectors.all()
    }
}

File: fixtures/scale/csharp/large/Families/Atbash/AtbashDescriptor.cs
using System.Collections.Generic;
using CipherLab.Abstractions;

namespace CipherLab.Families.Atbash
{
    public class AtbashDescriptor : IFamilyDescriptor
    {
        public string Family => "atbash";
        public string Suite => "classical";
        public CipherKind Kind => CipherKind.Letter;

        public ICipher Cipher()
        {
            return new AtbashCipher();
        }

        public List<TestVector> Vectors()
        {
            return AtbashVectors.All();
        }
    }
}
